projects/example-app/src/app/books/reducers/collection.reducer.ts

```diff
@@ -16,54 +17,43 @@ const initialState: State = {
   ids: [],
 };
 
-export function reducer(
-  state = initialState,
-  action:
-    | SelectedBookPageActions.SelectedBookPageActionsUnion
-    | CollectionPageActions.CollectionPageActionsUnion
-    | CollectionApiActions.CollectionApiActionsUnion
-): State {
-  switch (action.type) {
-    case CollectionPageActions.loadCollection.type: {
-      return {
-        ...state,
-        loading: true,
-      };
-    }
-
-    case CollectionApiActions.loadBooksSuccess.type: {
-      return {
-        loaded: true,
-        loading: false,
-        ids: action.books.map(book => book.id),
-      };
-    }
-
-    case CollectionApiActions.addBookSuccess.type:
-    case CollectionApiActions.removeBookFailure.type: {
-      if (state.ids.indexOf(action.book.id) > -1) {
-        return state;
+export const reducer = createReducer<State>(
+  [
+    on(CollectionPageActions.loadCollection, state => ({
+      ...state,
+      loading: true,
+      blha: true,
+    })),
+    on(CollectionApiActions.loadBooksSuccess, (state, { books }) => ({
+      loaded: true,
+      loading: false,
+      ids: books.map(book => book.id),
+    })),
+    // Supports handing multiple types of actions
+    on(
+      CollectionApiActions.addBookSuccess,
+      CollectionApiActions.removeBookFailure,
+      (state, { book }) => {
+        if (state.ids.indexOf(book.id) > -1) {
```

Comment:
Nit, use `includes` here instead of `indexOf`?

Reply:
Unrelated to this PR - let's keep them focused.